Fix/upload plugin on automated transfer

sejas opened the pull request:
#### Description

Currently, there is an error when a user uploads a zip plugin on a simple site with Atomic compatibility. This PR fixes this issue #62928.

#### Proposed Changes

* Navigate to Thank you screen after uploading a zip plugin to a simple site.

#### Testing Instructions

* Grab any zip plugin. You can find one here: https://wordpress.org/plugins/
* On a simple site, buy a pro account 
* Navigate to Plugins > Upload
* Accept the notice about domain renaming
* Drop your zip plugin to start uploading it
* Wait until the process finishes
* Observe it finishes successfully and you see the thank you screen
* Go to Plugins > installed plugins and observe your installed plugin is active

#### Screenshots
Thank you Screen:
<img width="1904" alt="Screenshot 2022-06-09 at 11 42 24" src="https://user-images.githubusercontent.com/779993/172828877-046d5462-a824-4d4b-a3f9-046a52ea7266.png">

Screencast:



https://user-images.githubusercontent.com/779993/172831009-839bd72a-47aa-4584-a4fa-8e44952415bf.mov



---

- Fixes #62928

sejas committed:
Install Plugin: add redirection for upload flow simple

sejas committed:
Install Plugin: add comments to describe the redirection context

wojtekn commented on client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx:
```diff
@@ -232,17 +232,23 @@ const MarketplacePluginInstall = ( { productSlug }: MarketplacePluginInstallProp
+			// Default process
 			( installedPlugin && pluginActive ) ||
-			( atomicFlow && transferStates.COMPLETE === automatedTransferStatus )
+			// Tranfer to atomic using a marketplace plugin
```
There is a small typo in Tranfer -> Transfer here.

sejas replied:
Thanks for your finding @wojtekn.

I've been able to reproduce the `null` on the URL. It seems to happen when I upload a zip plugin to a site that is already atomic. I'll fix it and ask for your review again.
Thank you!!

I've confirmed that the original issue, uploading zip plugins affects private and public sites. So the issue is not related to the visibility of the site.
<img width="695" alt="Screenshot 2022-06-13 at 10 01 07" src="https://user-images.githubusercontent.com/779993/173318476-f1db55e7-897b-4493-80d7-fffe4212ac73.png">

wojtekn reviewed (commented):
@sejas I reviewed the code and I think that the change looks clean. Comments that explain conditions are making this code easier to understand 👍  I tested it on my local Calypso - the PR changed the behavior when the plugin installation fails - now it doesn't display "This URL should not be accessed directly. [...] error but displays the Thank You page.

I've tested the workflow a few times using Vimeo and Hello Dolly plugin. I was able to go through successfully through the process two times - one for Hello Dolly, and one for Vimeo. In both cases, I've got a proper Thank You page URL, with a plugin slug in the URL. I was able to use the button to go to the "Manage Plugin" page and my plugin was listed there.

However, a few times I've got a Thank You page that was looking like a false result - the page showed that plugin was installed, but instead of the plugin slug in the URL, I saw `null` in one case and `false` in another. In both cases, the plugin was not installed. I also spot the case in which my site was not migrated to Atomic, but I've got that Thank You page displayed. The links looked as follows:

http://calypso.localhost:3000/marketplace/thank-you/null/<WP SITE SLUG\>.wordpress.com
http://calypso.localhost:3000/marketplace/thank-you/false/<WP SITE SLUG\>.wordpress.com

Did you have a similar experience? Is it that we are only changing error handling when plugin installation occurs, and not the actual error that leads to plugin installation failure?

Also, should we clarify the testing steps and say that the site should be Private? Or is it not relevant?

sejas committed:
Install Plugin: fix typo in comments

sejas committed:
Install Plugin: add restriction to avoid cache in automatedTransferStatus

sejas commented:
@wojtekn, I've found that the new issue you've seen was related to a recently transferred site.
The useSelector is returning the latest value of the transfer until you refresh the page, in this case, `complete`.
I've changed the conditions to trigger the redirection to thank you.

Now everything should work fine. Can you review it again?

I've fixed the typo in the comment too.

Thank you!

wojtekn reviewed (approved):
@sejas thanks. Change looks good. I've tested it for both plugins I used before, and it worked fine each time.
